app/views/registrations/register.html.erb

```diff
@@ -31,13 +31,13 @@
       <% end %>
     <% end %>
   <% else %>
-    <% if @competition.uses_new_registration_service? %>
+    <% user_may_register = current_user.cannot_register_for_competition_reasons(@competition).length == 0 %>
+    <% if @competition.uses_new_registration_service? && user_may_register && ( @competition.registration_opened? || @competition.user_can_pre_register?(current_user)) %>
```

Comment:
This is quite a lot of combined checks. Can we move them either into the controller or (better yet!) into the model?

Reply:
something like `@competition.should_show_register_v2_panel?(current_user) ` ?